src/test/java/com/sonatype/insight/brain/cli/CommandLineOptionFilterTest.groovy

```diff
@@ -0,0 +1,142 @@
+ * Includes the third-party code listed at http://links.sonatype.com/products/nexus/attributions.
+ * "Sonatype" is a trademark of Sonatype, Inc.
+ */
+package com.sonatype.insight.brain.cli
+
+import groovy.cli.commons.CliBuilder
+
+import spock.lang.Specification
+
+class CommandLineOptionFilterTest
+    extends Specification
+{
+  def cli
+
+  def 'setup'() {
+    cli = new CliBuilder()
+    cli.with {
+      h(longOpt: 'help', '')
+      a(longOpt: 'authentication', '', args: 1)
+      i(longOpt: 'application-id', '', args: 1)
+      s(longOpt: 'server-url', '', args: 1)
+      e(longOpt: 'ignore-system-errors', '')
+      r(longOpt: 'result-file', '', args: 1)
+      t(longOpt: 'stage', '', args: 1)
+      X(longOpt: 'debug', '')
+      xc(longOpt: 'expanded-coverage', '')
+      w(longOpt: 'fail-on-policy-warnings', '')
+      p(longOpt: 'proxy', '', args: 1)
+      U(longOpt: 'proxy-user', '', args: 1)
+      _(longOpt: 'pki-authentication', '')
+    }
+  }
+
+  def 'not letting help and pki options thru'() {
+    setup:
+      String[] params = [
+          '-h',
+          '-a', 'auth',
+          '-i', 'appId',
+          '-s', 'serverUrl',
+          '-e',
+          '-t', 'develop',
+          '-X',
+          '-xc',
+          '-w',
+          '--pki-authentication',
+          '-p', 'proxyUser',
+          '-U', 'proxy:creds',
+          '-r', 'out.json',
+          'dir1', 'file1', '-h'
+      ]
+
+    when:
+      def args = CommandLineOptionFilter.filterCommandLineParams(params)
+      def options = cli.parse(args)
+
+    then:
+      !options.h
+      !options.'help'
+      !args.contains('-h')
+      !args.contains('--help')
+      args.contains('-a')
+      options.a == 'auth'
+      args.contains('-i')
+      options.i == 'appId'
+      args.contains('-s')
+      options.s == 'serverUrl'
+      args.contains('-e')
+      args.contains('-t')
+      options.t == 'develop'
+      args.contains('-X')
+      args.contains('-xc')
+      args.contains('-w')
+      !args.contains('--pki-authentication')
+      args.contains('-p')
+      options.p == 'proxyUser'
+      args.contains('-U')
+      options.U == 'proxy:creds'
+      args.contains('-r')
+      options.r == 'out.json'
+  }
+
+  def 'test shortened param names'() {
+    setup:
+      String[] params = [
+          '--help',
+          '--authentication', 'u:p',
+          '--server-url', 'http://myServer',
+          '--application-id', 'myApp',
+          '--ignore-system-errors',
+          '--result-file', 'file.json',
+          '--stage', 'develop',
+          '--debug',
+          '--expanded-coverage',
+          '--fail-on-policy-warnings',
+          '--proxy', 'myProxy',
+          '--proxy-user', 'user:pass',
+          '--pki-authentication',
+          'myArtifact.jar'
+      ]
+
+    when:
+      def args = CommandLineOptionFilter.filterCommandLineParams(params)
+      def options = cli.parse(args)
+
+    then:
+      args.contains('-a')
```

Comment:
a suggestion to make this a bit cleaner and help future comprehension, we can include hamcrest-all and reduce the args check to something more like this:

> when:
      def args = CommandLineOptionFilter.filterCommandLineParams(params).toList()
      def options = cli.parse(args)
    then:
      args containsInAnyOrder(
          '-a', 'auth',
          '-i', 'appId',
          '-s', 'serverUrl',
          '-e',
          '-t', 'develop',
          '-X',
          '-xc',
          '-w',
          '-p', 'proxyUser',
          '-U', 'proxy:creds',
          '-r', 'out.json',
          'dir1', 'file1'
      )

Reply:
I don't think that would catch a programming error where the value for '-a' was actually assigned to '-t', for example.  That's why I was checking args for existence and options for the resolved values.